storage/factory/pruningStorerFactory.go

```diff
@@ -354,9 +367,48 @@ func (psf *StorageServiceFactory) CreateForMeta() (dataRetriever.StorageService,
 	store.AddStorer(dataRetriever.StatusMetricsUnit, statusMetricsStorageUnit)
 	store.AddStorer(dataRetriever.TxLogsUnit, txLogsUnit)
 
+	historyTxUnit, hashEpochUnit, err := psf.createHistoryStorersIfNeeded()
+	if err != nil {
+		return nil, err
+	}
+
+	if psf.generalConfig.FullHistory.EnableHistoryNode {
+		successfullyCreatedStorers = append(successfullyCreatedStorers, historyTxUnit)
+		store.AddStorer(dataRetriever.TransactionHistoryUnit, historyTxUnit)
+
+		successfullyCreatedStorers = append(successfullyCreatedStorers, hashEpochUnit)
+		store.AddStorer(dataRetriever.HashEpochUnit, hashEpochUnit)
+	}
+
 	return store, err
 }
 
+func (psf *StorageServiceFactory) createHistoryStorersIfNeeded() (*pruning.PruningStorer, *storageUnit.Unit, error) {
+	if !psf.generalConfig.FullHistory.EnableHistoryNode {
```

Comment:
Can we perhaps log the creation of these storers? (log.Debug)

Reply:
for the other storers there are not logs for creation